src/compiler/checker.ts

```diff
@@ -13197,10 +13197,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         else if (!member.symbol.isReplaceableByMethod) {
             symbol.declarations.push(member);
         }
-        if (symbolFlags & SymbolFlags.Value) {
-            if (!symbol.valueDeclaration || symbol.valueDeclaration.kind !== member.kind) {
-                symbol.valueDeclaration = member;
-            }
+        if (symbolFlags & SymbolFlags.Value && !symbol.valueDeclaration) {
+            symbol.valueDeclaration = member;

```

Comment:
this isn't crucial for the fix but when investigating this issue I found those lines to be quite suspicious and there are no tests that would prove the old logic to be necessary